### PR TITLE
always run finally blocks no matter how we leave the try block (RT #112099)

### PR DESCRIPTION
**Pull request by mauke**

### CODE REVIEW

**Comment by karenetheridge** [comment]
https://rt.cpan.org/Ticket/Display.html?id=112099

**Comment by karenetheridge** [comment]
I'll pass this around for people to eyeball today, and if there are no screaming objections, send it out as -TRIAL tonight.

**Comment by mauke** [comment]
Cool, thanks. Note that this does change behavior a bit: `try { exit } finally { ... }` now runs `...` whereas it didn't before.

**Comment by shadowcat-mst** [comment]
I ... think ideally, if a user does that, a small gnome exits the monitor and hits the developer with a mallet.

Also, that would quite possibly fix a bug in CGI scripts compiled to be persistent with an overriden exit.

So, on the whole, LGTM

File: lib/Try/Tiny.pm [hunk 1]
```diff
@@ -29,6 +29,8 @@ BEGIN {
   *_HAS_SUBNAME = ($su || $sn) ? sub(){1} : sub(){0};
 }
 
+our @_finally_guards;
```
Comment (ribasushi) [inline lib/Try/Tiny.pm]: `our @some_variable` is generally perceived as an invite of _please feel free to fondle me if you think it is justified_. Even (or often especially) if rhe variable is `_` prefixed.

In the case of `T::T` there is no scenario where such "fondling" is justified, therefore a `my %stackstate` with `local()` on a key is a better solution.

Comment (mauke) [inline lib/Try/Tiny.pm]: I don't think that's going to be a problem in practice. Try::Tiny already contains the following fondleables: `$VERSION`, `@EXPORT`, `_subname`, `_HAS_SUBNAME`, and an entire hidden package: `Try::Tiny::ScopeGuard`.

**Comment by karenetheridge** [comment]
Thanks! merged and released as 0.25-TRIAL.

**Pull request closed without merging**